File: code/86202.pas
program Bai202;
  var n, a: integer;
  function SNT(x: longint) : boolean;
    var i :longint;
    begin
    if n < 2 then exit(false);
    for i := 2 to x div 2 do 
        if x mod i = 0 then exit(false);
        exit(true);
    end;
begin
  readln(n);
  for a := 2 to n div 2 do
      if (SNT(a) = true) and (SNT(n - a)) = true then
  writeln (a, '   ', n-a);
end.

File: code/86202.cpp
#include <iostream>
using namespace std;

bool snt(int n){
    int x;
    cin>>n;
    if (n<2)
        return false;
    for (x=2; x<=n-1; x++)
        if (n%x==0)
            return false;
    return true;
}

int main() 
{
    int n,a;
    cin>>n;
    for (a=2; a<=n/2; a++){
        if (snt(a) && snt(n-a))
            cout<<a<<" "<<n-a<<endl;
    }
    return 0;
}